src/core/string_map_test.cc

```diff
@@ -117,4 +117,17 @@ TEST_F(StringMapTest, Ttl) {
   EXPECT_TRUE(it == sm_->end());
 }
 
+TEST_F(StringMapTest, ReallocIfNeeded) {
+  for (size_t i = 0; i < 100; i++)
```

Comment:
yes, this setup is not enough to force reallocations. 
you must understand how mimalloc works to do this.

Reply:
I know the page sizes, but I don't test here for results, I test for implementation correctness. This is why I have the 1.0 condition

Reply:
So I should allocate lots of values and then delete 90% and check it contracted?

Reply:
basically you need lots of allocations (dozens opf thousands) that span several 64K pages.
then you delete 80% of them uniformly (i.e. skip deletion of every 5th element).
Now all mimalloc pages still exist but are under-utilized and ReallocIfNeeded should trigger for most entries.
using mi_heap_visit_blocks with mi_heap_get_default you can confirm that number of pages was indeed reduced

Reply:
why trigger for most but not all? because we do not realloc from a page that is currently "active", i.e. allocations use it until it gets depleted.

Reply:
here is the explanation on how mimalloc works: https://www.microsoft.com/en-us/research/uploads/prod/2019/06/mimalloc-tr-v1.pdf